src/ProgressOnderwijsUtils/Win32/VirusScan.cs

```diff
@@ -0,0 +1,55 @@
+﻿#nullable disable annotations
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.Windows.Sdk;
+using ProgressOnderwijsUtils;
+
+namespace ProgressOnderwijsUtils.Win32
+{
+    public static class VirusScan
+    {
+        public static bool IsMalware(string content, string contentName)
+        {
+            using var scanSession = new VirusScanSessie("Progress.Net");
+            PInvoke.AmsiScanString(scanSession.Context, content, contentName, IntPtr.Zero, out var result);
+            return ResultIsMalware(result);
+        }
+
+        public static bool IsMalware(byte[] buffer, string contentName) => IsMalwareUnsafe(buffer, contentName);
+
+        static unsafe bool IsMalwareUnsafe(byte[] buffer, string contentName)
+        {
+            var sizet = Marshal.SizeOf(typeof(byte)) * buffer.Length;
+            var bufferPtr = Marshal.AllocHGlobal(sizet).ToPointer();
+
+            using var scanSession = new VirusScanSessie("Progress.Net");
+            PInvoke.AmsiScanBuffer(scanSession.Context, bufferPtr, (uint)buffer.LongLength, contentName, IntPtr.Zero, out var result);
+            return ResultIsMalware(result);
+        }
+
+        static bool ResultIsMalware(AMSI_RESULT result)
+        {
+            // Er is een gedocumenteerde methode AmsiResultIsMalware, maar deze is niet daadwerkelijk beschikbaar
+            if (result == AMSI_RESULT.AMSI_RESULT_CLEAN) {
+                return false;
+            } else if (result == AMSI_RESULT.AMSI_RESULT_NOT_DETECTED) {
+                return false;
+            } else if (result >= AMSI_RESULT.AMSI_RESULT_BLOCKED_BY_ADMIN_START && result <= AMSI_RESULT.AMSI_RESULT_BLOCKED_BY_ADMIN_END) {
+                throw new("Scannen is geblokkeerd");
+            }
+            return true;
+        }
+    }
+
+    public sealed class VirusScanSessie : IDisposable
+    {
+        public readonly IntPtr Context;
+
+        public VirusScanSessie(string appName)
+            => PInvoke.AmsiInitialize(appName, out Context);
+
+        public void Dispose()
+            => PInvoke.AmsiUninitialize(Context);

```

Comment:
needs a finalizer, almost certainly.

Reply:
Deze class heeft niet veel toegevoegde waarde en kan wel weg lijkt me.

Reply:
Anders kun je inheriten van SafeHandle.

Reply:
ik weet even niet wat hier nu mee te doen

Reply:
Ik denk dat het makkelijkste is de class weg te maken en AmsiUninitialize aan te roepen in een finally.

Reply:
@chrarnoldus zoiets?

Reply:
Ja; en er is nog de Release op IShellItem2.